aea/helpers/dialogue/base.py

```diff
@@ -412,90 +395,145 @@ def last_message(self) -> Optional[Message]:
 
         return last_message
 
-    def get_message(self, message_id_to_find: int) -> Optional[Message]:
+    @property
+    def is_empty(self) -> bool:
+        """
+        Check whether the dialogue is empty.
+
+        :return: True if empty, False otherwise
+        """
+        return len(self._outgoing_messages) == 0 and len(self._incoming_messages) == 0
+
+    def counterparty_from_message(self, message: Message) -> Address:
+        if self.is_message_by_self(message):
+            counterparty = message.to
+        else:
+            counterparty = message.sender
```

Comment:
One liner please where possible:
```suggestion
        counterparty = message.to if self.is_message_by_self(message) else  message.sender
```

Reply:
Missing docstrings

Reply:
Done